Test/Integration/Category/CategoryCacheTest.php

```diff
@@ -0,0 +1,341 @@
+        }
+    }
+
+    /**
+     * @dataProvider getCategoryProvider
+     * @depends      testCategoryPlpMissBackendRenderOff
+     * @magentoConfigFixture current_store system/full_page_cache/caching_application 1
+     * @magentoConfigFixture current_store algoliasearch_advanced/advanced/prevent_backend_rendering 1
+     * @magentoConfigFixture current_store algoliasearch_instant/instant/replace_categories 1
+     * @magentoDataFixture Algolia_AlgoliaSearch::Test/Integration/_files/backend_render_user_agents.php
+     * @magentoCache full_page enabled
+     */
+    public function testCategoryPlpHitBackendRenderWhiteList(int $categoryId, string $name): void
+    {
+        $config = $this->_objectManager->get(ScopeConfigInterface::class);
+        $preventBackend = $config->getValue('algoliasearch_advanced/advanced/prevent_backend_rendering', ScopeInterface::SCOPE_STORE);
+        $this->assertEquals(1, $preventBackend,"Prevent backend rendering must be enabled for this test.");
+
+        $testUserAgent = "Foobot";
+        $whitelist = $this->config->getValue('algoliasearch_advanced/advanced/backend_rendering_allowed_user_agents', ScopeInterface::SCOPE_STORE);
+        $this->assertStringContainsString($testUserAgent, $whitelist, "Allowed user agents for backend render must include $testUserAgent");
+
+        $this->registerPageHitSpy();
+
+        $_SERVER['HTTP_USER_AGENT'] = $testUserAgent;
+
+        $this->dispatchCategoryPlpRequest($categoryId);
+        $response = $this->getResponse();
+        $this->assertEquals(200, $response->getHttpResponseCode(), 'Request failed');
+    }
+
+    public function testFullPageCacheAvailable(): void
+    {
+        $types = $this->cacheManager->getAvailableTypes();
+        $this->assertContains('full_page', $types);
+    }
+
+    /**
+     * The \Magento\TestFramework\TestCase\AbstractController::dispatch is flawed for this use case as it does not
+     * populate the URI which is used to build the cache key in \Magento\Framework\App\PageCache\Identifier::getValue
+     *
+     * This provides a workaround
+     *
+     * @param string $uri
+     * @return void
+     */
+    protected function dispatchHttpRequest(string $uri): void
+    {
+        $request = $this->_objectManager->get(\Magento\Framework\App\Request\Http::class);
+        $request->setDispatched(false);
+        $request->setUri($uri);
+        $request->setRequestUri($uri);
+        $this->_getBootstrap()->runApp();
+    }
+
+    /**
+     * It is imperative to always use the same URL format between MISS and HIT to ensure
+     * that the cache key is generated consistently
+     * @param int $categoryId
+     * @return string
+     */
+    protected function getCategoryUrl(int $categoryId): string
+    {
+        return $this->url . $categoryId;
+    }
+
+    protected function dispatchCategoryPlpRequest(int $categoryId): void
+    {
+        $this->dispatchHttpRequest($this->getCategoryUrl($categoryId));
+    }
+
+    /**
```

Comment:
I didn't understood that the whole goal of this method was to ensure we get a HIT, maybe this is worth adding it in the description ?